### PR TITLE
Iam roles

### PR DESCRIPTION
**Pull request by max-jacobs**
Open sourcing the iam roles for Beekeeper

### CODE REVIEW

File: main/variables.tf [hunk 1]
```diff
@@ -203,3 +208,8 @@ variable "db-password-key" {
   description = "Key to acquire the database password for the strategy specified."
   type        = "string"
 }
+
+variable "cleanup-buckets" {
```
Comment (patduin) [inline main/variables.tf]: plural what's the format? comma separated list of buckets? s3:// or without?

Comment (max-jacobs) [reply main/variables.tf]: comma separated list of bucket arns, will update the description

Comment (patduin) [reply main/variables.tf]: ta

**Comment by abhimanyugupta07** [comment]
Can you also add a CHANGELOG to the project ?

File: main/cloudwatch.tf [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+             "title": "Shunting Yard SQS Queue Size Metrics"
+           }
+       },
+       {
+          "type":"metric",
+          "width":12,
+          "height":6,
+          "properties":{
+             "metrics": [
+               [ "AWS/SQS", "NumberOfMessagesDeleted", "QueueName", "${local.instance_alias}-sqs-queue" ]
```
Comment (givanovexpe) [inline main/cloudwatch.tf]: ${local.instance_alias}-sqs-queue -> ${var.queue_name}

File: main/cloudwatch.tf [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+             "title": "Shunting Yard SQS Deleted Messages"
+          }
+       },
+       {
+          "type":"metric",
+          "width":12,
+          "height":6,
+          "properties":{
+             "metrics": [
+                 [ "AWS/SQS", "ApproximateAgeOfOldestMessage", "QueueName", "${local.instance_alias}-sqs-queue" ]
```
Comment (givanovexpe) [inline main/cloudwatch.tf]: ${local.instance_alias}-sqs-queue -> ${var.queue_name}

File: main/cloudwatch.tf [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      namespace   = "AWS/ECS"
+      metric_name = "CPUUtilization"
+      threshold   = "90"
+    },
+    {
+      alarm_name  = "${local.instance_alias}-memory"
+      namespace   = "AWS/ECS"
+      metric_name = "MemoryUtilization"
+      threshold   = "80"
+    }
```
Comment (givanovexpe) [inline main/cloudwatch.tf]: Actually we need 4:

    {
      alarm_name  = "${local.instance_alias}-path-scheduler-cpu"
      namespace   = "AWS/ECS"
      metric_name = "CPUUtilization"
      threshold   = "90"
    },
    {
      alarm_name  = "${local.instance_alias}-path-scheduler-memory"
      namespace   = "AWS/ECS"
      metric_name = "MemoryUtilization"
      threshold   = "80"
    }
    {
      alarm_name  = "${local.instance_alias}-cleanup-cpu"
      namespace   = "AWS/ECS"
      metric_name = "CPUUtilization"
      threshold   = "90"
    },
    {
      alarm_name  = "${local.instance_alias}-cleanup-memory"
      namespace   = "AWS/ECS"
      metric_name = "MemoryUtilization"
      threshold   = "80"
    }

File: main/cloudwatch.tf [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+  dimensions = [
+    {
+      ClusterName = "${local.instance_alias}"
+      ServiceName = "${local.instance_alias}-service"
+    },
+    {
+      ClusterName = "${local.instance_alias}"
+      ServiceName = "${local.instance_alias}-service"
+    }
+  ]
```
Comment (givanovexpe) [inline main/cloudwatch.tf]: we need 4:

    {
      ClusterName = "${local.instance_alias}"
      ServiceName = "${local.instance_alias}-path-scheduler-service"
    },
    {
      ClusterName = "${local.instance_alias}"
      ServiceName = "${local.instance_alias}-path-scheduler-service"
    }
    {
      ClusterName = "${local.instance_alias}"
      ServiceName = "${local.instance_alias}-cleanup-service"
    },
    {
      ClusterName = "${local.instance_alias}"
      ServiceName = "${local.instance_alias}-cleanup-service"
    }

File: main/cloudwatch.tf [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+data "template_file" "sqs_widgets" {
+  template = <<EOF
+       {
+          "type":"metric",
+          "width":12,
+          "height":6,
+          "properties":{
+             "metrics": [
+                 [ "AWS/SQS", "NumberOfMessagesSent", "QueueName", "${local.instance_alias}-sqs-queue" ],
+                 [ "AWS/SQS", "NumberOfMessagesReceived", "QueueName", "${local.instance_alias}-sqs-queue" ]
```
Comment (givanovexpe) [inline main/cloudwatch.tf]: ${local.instance_alias}-sqs-queue -> ${var.queue_name}

File: main/cloudwatch.tf [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+       },
+       {
+          "type":"metric",
+          "width":12,
+          "height":6,
+          "properties":{
+        	 "metrics": [
+               [ "AWS/SQS", "ApproximateNumberOfMessagesVisible", "QueueName", "${local.instance_alias}-sqs-queue" ],
+               [ "AWS/SQS", "ApproximateNumberOfMessagesDelayed", "QueueName", "${local.instance_alias}-sqs-queue" ],
+               [ "AWS/SQS", "ApproximateNumberOfMessagesNotVisible", "QueueName", "${local.instance_alias}-sqs-queue" ]
```
Comment (givanovexpe) [inline main/cloudwatch.tf]: ${local.instance_alias}-sqs-queue -> ${var.queue_name}

File: main/templates.tf [hunk 8]
```diff
@@ -1,60 +1,75 @@
...
+/**
+ * Copyright (C) 2018-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ */
+
+data "template_file" "beekeeper_path_scheduler_container_definition" {
+  template = "${file("${path.module}/files/container-definition.json")}"
 
   vars {
```
Comment (max-jacobs) [inline main/templates.tf]: @givanovexpe hey, is it okay if I change all the template file vars blocks to this:

```
vars = { ...
```

getting this error in egdl plans:

```
Error: Unsupported block type
--
on .terraform/modules/beekeeper/main/templates.tf line 30, in data "template_file" "beekeeper_path_scheduler_config":
30:   vars {

Blocks of type "vars" are not expected here. Did you mean to define argument
"vars"? If so, use the equals sign to assign it a value.
```

Comment (max-jacobs) [reply main/templates.tf]: looks like terraform 11 specifies no equals (https://www.terraform.io/docs/configuration-0-11/interpolation.html#templates) but terraform 12 requires it.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+## 7. Submit a pull request and describe the change
+
+Push your changes to your branch and open a pull request against the parent repo on GitHub. The project administrators will review your pull request and respond with feedback.
+
+# How Your Contribution Gets Merged
+
+Upon Pull Request submission, your code will be reviewed by the maintainers. They will confirm at least the following:
+
+- Tests run successfully (unit, coverage, integration, style).
```
Comment (abhimanyugupta07) [inline CONTRIBUTING.md]:
```suggestion
```

File: main/README.md [hunk 10]
```diff
@@ -1,42 +1,58 @@
...
+| graphite\_port | Graphite port. | string | `"2003"` | no |
+| graphite\_prefix | Prefix for Graphite metrics. | string | n/a | yes |
+| beekeeper\_tags | A map of tags to apply to resources. | map | n/a | yes |
+| dry\_run\_enabled | Enable to perform dry runs of deletions only. | string | `"false"` | no |
+
+# Contact
+
+## Mailing List
+If you would like to ask any questions about or discuss Apiary please join our mailing list at
+
```
Comment (abhimanyugupta07) [inline main/README.md]:
```suggestion
https://groups.google.com/forum/#!forum/apiary-user
```

File: main/iam-policy-sqs.tf [hunk 11]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (C) 2018-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ */
+
+resource "aws_iam_policy" "beekeeper_sqs" {
```
Comment (abhimanyugupta07) [inline main/iam-policy-sqs.tf]: I personally like to have policy, role and attachment together in one file `IAM-sqs.tf` but if this is the pattern we are following in this project, I am fine with it.

**Pull request merged**